clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslClientCallbackHandler.java

```diff
@@ -78,10 +82,21 @@ public void handle(Callback[] callbacks) throws UnsupportedCallbackException {
                 if (ac.isAuthorized())
                     ac.setAuthorizedID(authzId);
             } else if (callback instanceof ScramExtensionsCallback) {
+                if (SaslConfigs.GSSAPI_MECHANISM.equals(mechanism))
+                    return; // extensions are not supported for GSSAPI
```

Comment:
Why was this change made? I think the single `if` statement is better than returning here.

Reply:
Personal preference. I find this more readable than a bigger if check. Changed back to one `if` and now checking for the correct mechanism in each callback